src/components/Comments/Comment.js

```diff
@@ -111,6 +129,61 @@ const Comment = ({
     return COLORS.balance_gray;
   };
 
+  const username = comment?.data?.anon_user_info_color_name
+    ? `Anonymous ${comment.data?.anon_user_info_emoji_name}`
+    : user?.data?.username;
+
+  const onPressDM = async () => {
+    try {
+      setLoading({...loading, loadingDm: true});
+      if (!comment?.data?.anon_user_info_color_name) {
+        const channelName = [username, profile?.myProfile?.username].join(',');
+        const selectedUser = {
+          user: {
+            name: channelName,
+            image: comment?.user?.data?.profile_pic_url || DEFAULT_PROFILE_PIC_PATH
+          }
+        };
+        const members = [comment?.user?.id, profile.myProfile.user_id];
+        await createSignChat(members, selectedUser);
+      } else {
+        await sendMessageDM(comment?.id, 'comment', 'SIGNED');
+      }
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDm: false});
+    }
+  };
+
+  const onPressDMAnon = async () => {
+    try {
+      setLoading({...loading, loadingDmAnon: true});
+      await sendMessageDM(comment?.id, 'comment', 'ANONYMOUS');
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDmAnon: false});
+    }
+  };
+
+  const dataSheet = [
+    {
+      id: 1,
+      name: loading.loadingDm ? 'Loading...' : `Message ${username}`,
+      icon: <SendDMBlack />,
+      onPress: onPressDM
+    },
+    {
+      id: 2,
+      name: loading.loadingDmAnon ? 'Loading...' : `Message ${username} anonymously`,
+      icon: <SendDMAnonBlock />,
+      onPress: onPressDMAnon
+    }
+  ];
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for handling direct messages (DMs) and anonymous DMs is introduced here. The `onPressDM` and `onPressDMAnon` functions handle the sending of DMs, with appropriate loading states and error handling. The `dataSheet` array is used to configure the options in the `BottomSheetMenu`. However, it's important to ensure that the API calls made in these functions are correctly implemented and handle all possible edge cases.
```diff
+ const username = comment?.data?.anon_user_info_color_name
+   ? `Anonymous ${comment.data?.anon_user_info_emoji_name}`
+   : user?.data?.username;
+ 
+ const onPressDM = async () => {
+   try {
+     setLoading({...loading, loadingDm: true});
+     if (!comment?.data?.anon_user_info_color_name) {
+       const channelName = [username, profile?.myProfile?.username].join(',');
+       const selectedUser = {
+         user: {
+           name: channelName,
+           image: comment?.user?.data?.profile_pic_url || DEFAULT_PROFILE_PIC_PATH
+         }
+       };
+       const members = [comment?.user?.id, profile.myProfile.user_id];
+       await createSignChat(members, selectedUser);
+     } else {
+       await sendMessageDM(comment?.id, 'comment', 'SIGNED');
+     }
+   } catch (e) {
+     console.warn(e);
+   } finally {
+     refSheet.current.close();
+     setLoading({...loading, loadingDm: false});
+   }
+ };
+ 
+ const onPressDMAnon = async () => {
+   try {
+     setLoading({...loading, loadingDmAnon: true});
+     await sendMessageDM(comment?.id, 'comment', 'ANONYMOUS');
+   } catch (e) {
+     console.warn(e);
+   } finally {
+     refSheet.current.close();
+     setLoading({...loading, loadingDmAnon: false});
+   }
+ };
+ 
+ const dataSheet = [
+   {
+     id: 1,
+     name: loading.loadingDm ? 'Loading...' : `Message ${username}`,
+     icon: <SendDMBlack />,
+     onPress: onPressDM
+   },
+   {
+     id: 2,
+     name: loading.loadingDmAnon ? 'Loading...' : `Message ${username} anonymously`,
+     icon: <SendDMAnonBlock />,
+     onPress: onPressDMAnon
+   }
+ ];
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->